File: transpiler/javatests/com/google/j2cl/transpiler/integration/morebridgemethods/TestCase9366.java
package com.google.j2cl.transpiler.integration.morebridgemethods;

import jsinterop.annotations.JsType;

public class TestCase9366 {
  @SuppressWarnings("rawtypes")
  static class B<B1 extends Comparable> {
    @SuppressWarnings("unused")
    public String get(String value) {
      return "B get String";
    }
  }

  @JsType
  static class C<C1 extends String> extends B<C1> {
    @Override
    public String get(String value) {
      return "C get String";
    }
  }

  @SuppressWarnings({"unchecked", "rawtypes"})
  public static void test() {
    C c = new C();
    assert ((B) c).get("").equals("C get String");
    assert c.get("").equals("C get String");
  }
}

File: transpiler/javatests/com/google/j2cl/integration/kotlin/morebridgemethods/TestCase9366.kt
/*
 * Copyright 2022 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package morebridgemethods

import com.google.j2cl.integration.testing.Asserts.assertTrue
import jsinterop.annotations.JsType

object TestCase9366 {
  internal open class B<B1 : Comparable<String>> {
    open fun get(value: String): String {
      return "B get String"
    }
  }

  @JsType
  internal class C<C1 : String> : B<C1>() {
    override fun get(value: String): String {
      return "C get String"
    }
  }

  fun test() {
    val c = C<String>()
    assertTrue(c.get("") == "C get String")
    assertTrue(c.get("") == "C get String")
  }
}
